lib/DXIL/DxilShaderFlags.cpp

```diff
@@ -778,17 +840,86 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
   flag.SetSamplerDescriptorHeapIndexing(hasSamplerDescriptorHeapIndexing);
   flag.SetAtomicInt64OnHeapResource(hasAtomicInt64OnHeapResource);
   flag.SetAdvancedTextureOps(hasAdvancedTextureOps);
-  flag.SetWriteableMSAATextures(hasWriteableMSAATextures);
+  flag.SetWriteableMSAATextures(setWriteableMSAATexturesBasedOnGlobal
+                                    ? hasWriteableMSAATexturesGlobal
+                                    : hasWriteableMSAATextures);
```

Comment:
This check seems unnecessary and inconsistent as there's already a place above that will set `hasWriteableMSAATextures` to true if `setWiteableMSAATexturesBasedOnGlobal` is true and there are any RWMS textures in the module. 

Is there a case where there aren't RWMS textures in the module and yet there are somehow handle creation calls for a RWMS texture? That seems the only case where `hasWriteableMSAATexturesGlobal` would be false, but somehow `hasWriteableMSAATextures` would be true and we want the false because `setWiteableMSAATexturesBasedOnGlobal` is set. 

Said another way, I thinkt hat if `hasWriteableMSAATextures` is true, then `hasWriteableMSAATexturesGlobal` _must_ be true, so I don't know why you wouldn't just use `hasWriteableMSAATextures` here without any contidional. 
```suggestion
  flag.SetWriteableMSAATextures(hasWriteableMSAATextures);
```

Reply:
It *is* possible to use a dynamic resource so `hasWriteableMSAATextures` can be true with `hasWriteableMSAATexturesGlobal` being false.

However, revisiting this, I think the compatibility code won't quite get it right either.  Before, the flag would be set if either there was a global resource, or if [TextureStoreSample was used](https://github.com/microsoft/DirectXShaderCompiler/pull/6207/files#diff-d35f0d00e33e546a948baeacb7cbd7b6e5d3ebf0650bf93bd354e0f47821c20aR710).  But there are more potential uses of the resource than `TextureStoreSample`, so I added code to catch any other use.  However, since the `TextureStoreSample` isn't counted separately now, neither flag captures the exact behavior of 1.7.

I think I'm going to rename `hasWriteableMSAATexturesGlobal` to `hasWriteableMSAATextures_1_7`, then set this based on global, and set it when `TextureStoreSample` is encountered, then switch between the flags based on that.  Looks like another subtle test case is needed, ugh..